UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/FulfilmentServiceJob.cs

```diff
@@ -13,25 +18,77 @@ namespace UKHO.ExchangeSetService.FulfilmentService
+                await fulFilmentDataService.CreateExchangeSet(fulfilmentServiceQueueMessage, currentUtcDateTime);
 
-            logger.LogInformation(EventIds.CreateExchangeSetRequestCompleted.ToEventId(), "Create Exchange Set web job completed for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", fulfilmentServiceQueueMessage.BatchId, fulfilmentServiceQueueMessage.CorrelationId);
+                logger.LogInformation(EventIds.CreateExchangeSetRequestCompleted.ToEventId(), "Create Exchange Set web job completed for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", fulfilmentServiceQueueMessage.BatchId, fulfilmentServiceQueueMessage.CorrelationId);
+            }
+            catch (Exception ex)
+            {
+                EventId exceptionEventId = EventIds.SystemException.ToEventId();
+
+                if (ex.GetType() == typeof(FulfilmentException))
+                    exceptionEventId = ((FulfilmentException)ex).EventId;
+
+                FulfilmentException fulfilmentException = new FulfilmentException(exceptionEventId);
+                string errorMessage = string.Format(fulfilmentException.Message, exceptionEventId.Id, fulfilmentServiceQueueMessage.CorrelationId);
+
+                await CreateAndUploadErrorFileToFileShareService(fulfilmentServiceQueueMessage, exceptionEventId, errorMessage, batchFolderPath);
+
+                if (ex.GetType() != typeof(FulfilmentException))
+                    logger.LogError(exceptionEventId, "Unhandled exception while processing Exchange Set web job for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId} and Exception:{Exception} and StackTrace:{StackTrace}", fulfilmentServiceQueueMessage.BatchId, fulfilmentServiceQueueMessage.CorrelationId, ex.Message, ex.StackTrace);
+            }
+        }
+
+        public async Task CreateAndUploadErrorFileToFileShareService(SalesCatalogueServiceResponseQueueMessage fulfilmentServiceQueueMessage, EventId eventId, string errorMessage, string batchFolderPath)
+        {
+            fileSystemHelper.CheckAndCreateFolder(batchFolderPath);
+
+            var errorFileFullPath = Path.Combine(batchFolderPath, fileShareServiceConfig.Value.ErrorFileName);
+            fileSystemHelper.CreateFileContent(errorFileFullPath, errorMessage);
+
+            if (fileSystemHelper.CheckFileExists(errorFileFullPath))
+            {
+                var isUploaded = await fileShareService.UploadFileToFileShareService(fulfilmentServiceQueueMessage.BatchId, batchFolderPath, fulfilmentServiceQueueMessage.CorrelationId, fileShareServiceConfig.Value.ErrorFileName);
+
+                if (isUploaded)
+                {
+                    logger.LogError(eventId, "Error while processing Exchange Set creation and error file created for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", fulfilmentServiceQueueMessage.BatchId, fulfilmentServiceQueueMessage.CorrelationId);
```

Comment:
Kindly change eventid to EventIds.ErrorTxtIsUploaded.ToEventId() and also log event id in logger message

logger.LogError(EventIds.ErrorTxtIsUploaded.ToEventId(), "Error while processing Exchange Set creation and error file is created and uploaded in file share service with ErrorCode-EventId:{eventId} for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", eventid, fulfilmentServiceQueueMessage.BatchId, fulfilmentServiceQueueMessage.CorrelationId);

Reply:
Resolved